### PR TITLE
KOA-5626 adopt new colors

### PR DESCRIPTION
**Pull request by gert-janvercauteren**
Adopt all new colours and update snapshots

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/Tests/UnitTests/BPKColorTest.m [hunk 1]
```diff
@@ -27,7 +27,7 @@ @interface BPKColorTest : XCTestCase
 
 @implementation BPKColorTest
 
-- (void)testColorSubsetExists {
+- (void)disabled_testColorSubsetExists {
```
Comment (marianeum) [inline Backpack/Tests/UnitTests/BPKColorTest.m:30]: should we remove those instead or do we need to rewrite them with semantic examples?

**Review by marianeum — APPROVED** [review]

**Review by olliecurtis — CHANGES_REQUESTED** [review]
[Calendar (Traffic Lights)](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/Backpack/Tests/SnapshotTests/ReferenceImages_64/BPKCalendarSnapshotTest/testCalendarWithCustomColorDates%402x.png): Are we ok with the purple not being visible at all in this example not sure what state that represents but just want to check?

[Flare](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/Backpack/Tests/SnapshotTests/ReferenceImages_64/BPKFlareViewSnapshotTest/testFlareViewBottomRoundedSnapshot%402x.png): Do we expect the pointer to be different colour to the body of the flare?

[Map Markers](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/Backpack/Tests/SnapshotTests/ReferenceImages_64/BPKMapAnnotationSnapshotTest/testMapAnnotationsWithIcons_dark_mode%402x.png): 

- Slight concern with the lighter blue for dark mode and its contrast on the white?
- Similarly the disabled state you can't see the text at all?

[TabBar](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/Backpack/Tests/SnapshotTests/ReferenceImages_64/BPKTabBarControllerSnapshotTest/testTabBarControllerWithIcons_dark_mode%402x.png): Looks like icons have gone missing?

**Review by olliecurtis — COMMENTED** [review]
[Switch (Dark Mode)](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/Backpack/Tests/SnapshotTests/ReferenceImages_64/Backpack_Unit_SnapshotTests.BPKSwitchSnapshotTest/testSwitchWhenOn_dark_mode%402x.png): Doesn't look to be using the dark mode as per the design

Dialog: The background to the icon looks to be a different colour to the main body, is this expected?
[Dark Mode](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/screenshots/iPhone%208-dialog___delete-confirmation_dm.png) 
[Light Mode](https://github.com/Skyscanner/backpack-ios/blob/463f598ac1ee3dde9b85685ff13e0bea6b99cefb/screenshots/iPhone%208-dialog___delete-confirmation_lm.png)

File: Backpack/Tests/UnitTests/BPKColorTest.m (hunk 1 above)
Comment (gert-janvercauteren) [reply Backpack/Tests/UnitTests/BPKColorTest.m:30]: I have removed the tests and created a ticket to follow-up on creating new colour related tests.

**Comment by gert-janvercauteren** [comment]
> 

- Calendar: Calendar now uses a different 'custom' color. It was set to the deprecated Abisko color
- FlareView, yes this is expected. We colour the innerview different from the tip so we can verify the 'user settable part' is up to spec. The original snapshot had this as well, the change now is that panjin has changed to be berry.
- MapAnnotationView: Migrated to a new snapshot testing library that handles layer colours better
- TabBar: Fixed test, setting up correct colorways.

**Comment by gert-janvercauteren** [comment]
> 

Switch: Update library to make sure dark mode works properly
Dialog: The icon use 'textInverse', the mismatch in color is expected here.

**Review by olliecurtis — APPROVED** [review]

**Pull request merged**